### PR TITLE
refactoring phenotype to disease

### PR DESCRIPTION
**Pull request by TereseBo**
This PR contains refactoring of disease parsing functions replacing "phenotype" with "disease" in order to differentiate between phenotypes which are diseases and those that are  phenotypic abnormality (HPO terms).

This PR is a fix for #4255

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Parsing of diseases should still be functional

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
On main branch after running "scout --demo update diseases -f ./disease_file_folder"
image

In DB: 
image

On this branch after performing the same action:

image

image

**Review by northwestwitch — APPROVED** [review]
This looks simple enough, just a renaming of variables 👍🏻

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 ### Changed
 - In the diagnoses page genes associated with a disease are displayed using hgnc symbol instead of hgnc id
 ### Fixed
+- Refactored code in parsing of omim files and parsing of disease to use "disease" instead of "phenotype" to differentiate from HPO terms
```
[the remark this answers is not in the document]
Comment (TereseBo) [reply CHANGELOG.md:14]: Updated accordingly

File: scout/parse/omim.py [hunk 2]
```diff
@@ -19,16 +19,16 @@ def parse_omim_line(line, header):
     return omim_info
 
 
-def parse_genemap2_phenotypes(phenotype_entry, mim_number=None):
+def parse_genemap2_diseases(phenotype_entry, mim_number=None):
     """Parse the Phenotype entry of a genemap2 line
```
[the remark this answers is not in the document]
Comment (TereseBo) [reply scout/parse/omim.py:23]: Updated accordingly

**Pull request merged**